### ISSUE TITLE
Example files not running properly

### ISSUE THREAD

**Issue by Huy-Ngo** [McSinyx/palace#61]
https://github.com/McSinyx/palace/blob/49072f101e159c0e27b53b98f9367cecf57acf0f/examples/palace-hrtf.py#L60-L69

I'm trying to run some already written example files, and encounters some errors when trying to run `palace-hrtf.py`, particularly related to optional arguments.

Run command: `python3 palace-hrtf.py <source file path>`

On line 60:

```python
  File "palace-hrtf.py", line 60, in play
    with Context(dev, attrs) as ctx, Source() as src:
  File "src/palace.pyx", line 995, in palace.Source.__init__
TypeError: __init__() takes exactly 1 positional argument (0 given)
```

Which, imo, doesn't make sense, since that argument is optional. Anyway, I gave `ctx` as an argument, since `None` would lead to `RuntimeError: Called context is not current`.

Then after this is `Decoder(filename)` (expected 2, got 1). I tried to give it `ctx` as the second argument, but it says `ctx` is a `str`:

```python
TypeError: argument 'context' has incorrect type (expected palace.Context, got str)
```

This is even weirder than the previous one.

I believe you have already run this before upload it here, but somehow it doesn't work on my machine.

TL;DR:
- Optional arguments turn out to be not optional
- `ctx` somehow is misinterpreted as `str`.

**Comment by McSinyx** [comment]
It's quite likely that this is integration issues (i.e. mismatch of palace version and the examples).  See GH-52 (sorry for not noticing such incompatible change).

**Comment by Huy-Ngo** [comment]
I have pulled from master and `pip install --upgrade palace` before trying this. probably because this wasn't released yet?

**Comment by McSinyx** [comment]
No it have not.  I don't mind if you release 0.0.13 right now tho.

**Comment by Huy-Ngo** [comment]
I [failed](https://gist.github.com/Huy-Ngo/a5fb33d6bee5aa98dd22debc9c1af642) to install with `pip3 install palace/` too.

**Comment by McSinyx** [comment]
> subprocess.CalledProcessError: Command '['cmake', '.']' returned non-zero exit status 1.

Can you run `cmake palace/` to see what causes the issue?  I don't think I've touched the build part the last few days.

**Comment by Huy-Ngo** [comment]
Doing `cmake palace/` is fine. Nothing strange happens.

**Comment by Huy-Ngo** [comment]
@9a24f0 Can you install and run it?

**Comment by McSinyx** [comment]
LMAO just checked got the same thing.  `./setup.py install --user` runs fine tho, for now use that.  Strange thing tox builds palace using pip and does not raise any error.

Edit: after `./setup clean --all` pip installs alright.  Strange.

Edit: nothing is strange here.   ~~The source tree is copied into a venv before running setup.py.~~  *WRONG: it's not copied unless PEP 517 is asked for.*  Maybe pip should tell setup.py to clean before build... hmmm...

**Comment by Huy-Ngo** [comment]
Ah it's alright now, thanks for that
